RELEASE_NOTES.md

```diff
@@ -0,0 +1,44 @@
+# Performance Optimizations
+## Intel Architecture Processors
+  * Improved fp16/bf16 softmax performance with relaxed [accumulation mode](https://oneapi-src.github.io/oneDNN/dev_guide_attributes_accumulation_mode.html#doxid-dev-guide-attributes-accumulation-mode).
+  * Added support and improved perfomance for fp8 matmul with bf16/fp16.
+
+## Intel Graphics Products
+  * Introduced initial optimizations for GPUs based on Xe3 architecture.
+  * Improved performance for convolution for Intel Arc Graphics for Intel Core Ultra processors (Series 2) (formerly Lunar Lake).
+  * Improved performance of the following subgraphs with Graph API
+    * Scaled dot-product Attention (SDPA) [with causal mask](https://oneapi-src.github.io/oneDNN/dev_guide_graph_sdpa.html#doxid-dev-guide-graph-sdpa)
+    * Scaled dot-product Attention (SDPA) [with compressed key and value](https://oneapi-src.github.io/oneDNN/dev_guide_graph_sdpa_compressed_kv.html#doxid-dev-guide-graph-sdpa-compressed-kv)
+## AArch64-based Processors
+
+# Functionality
+  * Introduced support for `select` algorithm in binary primitive. The functionality is optimized for Intel CPUs.
+  * Enabled support for matmul primitive with grouped quantization on weight along N dimension
+  * Graph API: new [`select`](https://oneapi-src.github.io/oneDNN/dev_guide_op_select.html), [`GenIndex`](https://oneapi-src.github.io/oneDNN/dev_guide_op_genindex.html) and [`GreaterEqual`](https://oneapi-src.github.io/oneDNN/dev_guide_op_greaterequal.html) operations.
+  * Introduced support for fp16/bf16 compressed weights in fp32 matmul on Intel CPUs.
+  * Introduced support for grouped scales and zero points in reorder primitive.
+  * Enabled support for 4d weight scale in matmul primitive.
+  * Introduced preliminary support for 4-bit floating-point data types `f4_e2m1` and `f4_e3m0` in matmul and reorder, as well as `e8m0` scales data type in matmul and reorder.
+  * [experimental] Extended microkernel API:
+		Introduced int4 quantization support.
+		Fpmath mode API
+# Usability
+  * With SYCL runtime, memory objects on CPU engine are now reference-counted and no more need to be explicitly kept alive by user for the duration of the primitive execution. This align memory object lifetime behavior on CPU and GPU engines.
+  * Improve verbose diagnostic to better identify issues during dispatching, primitive and kernel creation for CPU primitive and GPU (in case of OpenCL implementation) primitive implementations.
+  * Improve verbose diagnostic to simplify debugging of nGEN fallbacks.
```

Comment:
nGEN is an implementation detail. I would suggest calling out specific situations when new diagnostics is triggered.

+@avmanerikar

```suggestion
  * Improved verbose diagnostics for Intel GPU driver compatibility issues.
```